pkg/inspecttypes/dockercompat/dockercompat.go

```diff
@@ -375,3 +376,57 @@ func convertToNatPort(portMappings []gocni.PortMapping) (*nat.PortMap, error) {
 	}
 	return &portMap, nil
 }
+
+type IPAMConfig struct {
+	Subnet  string `json:"Subnet,omitempty"`
+	Gateway string `json:"Gateway,omitempty"`
+	// TODO: IPRange
+}
+
+type IPAM struct {
+	// Driver is omitted
+	Config []IPAMConfig `json:"Config,omitempty"`
+}
+
+// Network mimics a `docker network inspect` object.
+// From https://github.com/moby/moby/blob/v20.10.7/api/types/types.go#L430-L448
+type Network struct {
+	Name   string            `json:"Name"`
+	ID     string            `json:"Id,omitempty"` // optional in nerdctl
+	IPAM   IPAM              `json:"IPAM,omitempty"`
+	Labels map[string]string `json:"Labels"`
+	// Scope, Driver, etc. are omitted
+}
+
+func NetworkFromNative(n *native.Network) (*Network, error) {
```

Comment:
`error` doesn't seems to used here?

Reply:
True, but not new in this file. And a future version of this function may return an error.